libraries/chain/block_log.cpp

```diff
@@ -37,6 +37,8 @@ namespace eosio { namespace chain {
       uint32_t version         = 0;
       uint32_t first_block_num = 0;
       std::variant<genesis_state, chain_id_type> chain_context;
+      fc::path log_path; 
+      
```

Comment:
Just pass the log_path as a parameter to `block_log_preamble::read_from()` is enough. There's no reason the keep log_path as a member variable.